### DISCUSSION TITLE
Dont create a transaction when querying for data in v1.3

### DISCUSSION THREAD

**Discussion by yawhide** [sqlalchemy/sqlalchemy#7329] [Usage Questions] [answered]
Howdy,

Is there a way to not have sqlalchemy in `autocommit=False` mode create a transaction on the first query to the database in v1.3? I know in v1.4 the behaviour changes around this such that the first _write_ to the database starts a transaction in `autocommit=False` mode.

I cannot seem to find any documentation supporting that _but_ i haven't read through everything exhaustively. 

Thanks!

**Comment by zzzeek** [comment, marked as answer]
if by "create a transaction" you are looking at the actual transaction on the wire to the database, this is [the default behavior of any pep-249 database API](https://www.python.org/dev/peps/pep-0249/#commit), which is that it must emit BEGIN when the first SQL statement is encountered.  to turn this off you need to set DBAPI autocommit, which you can do as per the guidance at https://docs.sqlalchemy.org/en/14/core/connections.html#setting-transaction-isolation-levels-including-dbapi-autocommit  (those docs should be good for 1.3 also, it's just the 1.4 docs I've added some more clarifying verbiage)
